src/AbstractClient.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace CoverGenius\EbayRestPhpSdk;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Command\Guzzle\Description;
+use GuzzleHttp\Command\Guzzle\GuzzleClient;
+
+abstract class AbstractClient extends GuzzleClient
+{
+    /**
+     * EbayClient constructor. We setup our own constructor along with the parent
+     * GuzzleClient constructor to make this more testable. This will allow us
+     * to pass an instance of the Client with a mocked handler in our tests.
+     */
+    public function __construct(ClientInterface $client)
+    {
+        $baseUri = $client->getConfig('base_uri');
+
+        parent::__construct(
+            $client,
+            $this->buildDescription("$baseUri"),
```

Comment:
$baseUri meant to be a variable instead of a string inside the double quote here?

Reply:
@mith495 No, it's correct. 

$baseUri returns an instance of a class that has a `__toString()` magic method. So when the $baseUri is treated as a string, __toString will connect all the related uri segments and return a string.

Reply:
@mith495 Just for reference. This is the magic method Guzzle uses in its Uri class ...

```
    public function __toString()
    {
        return self::composeComponents(
            $this->scheme,
            $this->getAuthority(),
            $this->path,
            $this->query,
            $this->fragment
        );
    }
```